### PR TITLE
feat(npg): [CHK-1782] update commons library

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Update commons library

#### List of Changes

Add npgclient among the bean

#### Motivation and Context

Integration with new NPG

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Depends on pagopa/pagopa-infra#1233

### CODE REVIEW

File: helm/values-prod.yaml [hunk 1]
```diff
@@ -74,6 +78,7 @@ microservice-chart:
       PERSONAL_DATA_VAULT_API_KEY: personal-data-vault-api-key
       OTEL_EXPORTER_OTLP_HEADERS: elastic-otel-token-header
       ECOMMERCE_PAYMENT_METHODS_APY_KEY: payment-method-api-key
+      NPG_KEY: npg-key
```
Comment (giovanniberti) [inline helm/values-prod.yaml:81]:
```suggestion
      NPG_API_KEY: npg-api-key
```

File: helm/values-uat.yaml [hunk 2]
```diff
@@ -178,6 +183,10 @@ microservice-chart:
     OTEL_TRACES_SAMPLER: "always_on"
     ECS_SERVICE_NAME: pagopa-ecommerce-transactions-service
     ECS_SERVICE_ENVIRONMENT: "uat"
+    NPG_URI: "https://localhost/npg"
+    NPG_READ_TIMEOUT: 10000
+    NPG_CONNECTION_TIMEOUT: 10000
+    SPRING_MAX_IN_MEM_SIZE: 16777216
```
Comment (giovanniberti) [inline helm/values-uat.yaml:189]: Why do we need something different from the default value? (262 KB vs 16 MB here)

File: src/main/java/it/pagopa/transactions/configurations/NpgWebClientsConfig.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.netty.http.client.HttpClient;
+
+import java.util.concurrent.TimeUnit;
+
+@Configuration
+public class NpgWebClientsConfig {
+
+    @Value("${npg.client.maxInMemory}")
+    private int maxMemorySize;
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/configurations/NpgWebClientsConfig.java:22]: Is this codec configuration needed?

Comment (ciuffagianluca) [reply src/main/java/it/pagopa/transactions/configurations/NpgWebClientsConfig.java]: This configuration has been removed

**Review by infantesimone — DISMISSED** [review]

**Review by infantesimone — APPROVED** [review]

**Pull request merged**